### PR TITLE
Markduplicates

### PR DESCRIPTION
**Pull request by dmarron**
## PR checklist

Closes #[2245](https://github.com/nf-core/modules/issues/2245) 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Can we leave the unsorted bam test alone and just add the sorted cram test?

**Comment by dmarron** [comment]
> From @dmarron :
> 
> > we can't leave the unsorted bam alone, the test actually fails
> > for picard markduplicates
> > in the markduplicates documentation it says a sorted bam is required
> 
> Solid, good find! Let's add the test back and mark it as failing! https://pytest-workflow.readthedocs.io/en/stable/#test-options

I added this test back as succeeding, it turns out it could succeed if --ASSUME_SORT_ORDER queryname is passed in as a parameter to markduplicates

**Review by LaurenceKuhl — APPROVED** [review]
Hi,

looks good to me but adding an additional mandatory parameter might break all the pipelines/workflows currently using markduplicates, right?

File: tests/modules/nf-core/picard/markduplicates/main.nf [hunk 1]
```diff
@@ -2,21 +2,36 @@
...
+
+workflow test_picard_markduplicates_sorted_cram  {
+    input = [ [ id:'test', single_end:false ], // meta map
+              file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true)
+            ]
+    fasta = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+
+    PICARD_MARKDUPLICATES_SORTED_CRAM ( input, fasta, fai )
 }
```
Comment (Emiller88) [inline tests/modules/nf-core/picard/markduplicates/main.nf:37]:
```suggestion
}

// TODO Add a test with PICARD_MARKDUPLICATES ( input, [], [] )
```

**Review by Emiller88 — APPROVED** [review]
LGTM, just adding a comment for another test in the future!

**Review by qclayssen — APPROVED** [review]

**Pull request merged**